### PR TITLE
8255784: appcds/javaldr/ExceptionDuringDumpAtObjectsInitPhase.java test failed resulting in VM crash

### PR DESCRIPTION
**Pull request by yminqi**
Please review this simple change.
  The crash is caused by vm_exit_during_initialization which calls vm_exit, the latter will go check vm shutdown procedures. The shutdown checks thread and lock state, finds inconsistent state so the check fails. Here we could just use vm_direct_exit with some message as the process terminated.  vm_direct_exit just calls os::exit which quits without generating a coredump.

Tests: tier1-4

Thanks
Yumin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8255784](https://bugs.openjdk.java.net/browse/JDK-8255784): appcds/javaldr/ExceptionDuringDumpAtObjectsInitPhase.java test failed resulting in VM crash

### Reviewers
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to e1796d799b87ce25f7ac6afb27e3e8bd5ddbfc52
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1040/head:pull/1040`
`$ git checkout pull/1040`

### CODE REVIEW

File: src/hotspot/share/memory/archiveUtils.cpp [hunk 1]
```diff
@@ -323,8 +323,8 @@ void ArchiveUtils::log_to_classlist(BootstrapInfo* bootstrap_specifier, TRAPS) {
 void ArchiveUtils::check_for_oom(oop exception) {
   assert(exception != nullptr, "Sanity check");
   if (exception->is_a(SystemDictionary::OutOfMemoryError_klass())) {
-    vm_exit_during_cds_dumping(
-      err_msg("Out of memory. Please run with a larger Java heap, current MaxHeapSize = " SIZE_FORMAT "M",
-              MaxHeapSize/M));
+    vm_direct_exit(-1,
+      err_msg("Out of memory. Please run with a larger Java heap, current MaxHeapSize = "
+              SIZE_FORMAT "M", MaxHeapSize/M));
```
Comment (calvinccheung) [inline src/hotspot/share/memory/archiveUtils.cpp:328]: I didn't know the format string could be broken up into 2 lines.

**Review by calvinccheung — APPROVED** [review]
Looks good.

**Review by iklam — APPROVED** [review]

**Comment by yminqi** [comment]
@iklam @calvinccheung Thanks for review!

**Pull request closed without merging**